crossbeam-channel/src/select_macro.rs

```diff
@@ -686,9 +686,10 @@ macro_rules! crossbeam_channel_internal {
     ) => {{
         const _LEN: usize = $crate::crossbeam_channel_internal!(@count ($($cases)*));
         let _handle: &dyn $crate::internal::SelectHandle = &$crate::never::<()>();
+        let _entry = (_handle, 0, ::std::ptr::null());
 
         #[allow(unused_mut)]
-        let mut _sel = [(_handle, 0, ::std::ptr::null()); _LEN];
+        let mut _sel = [_entry; _LEN];
```

Comment:
I still think this change is unneeded and clippy lint's false positive.

Reply:
Thanks, I reverted this change and manually ignored `zero_repeat_side_effects` in that case.